src/realm/util/file.cpp

```diff
@@ -680,20 +680,50 @@ void File::resize(SizeType size)
 }
 
 
-void File::prealloc(SizeType offset, size_t size)
+void File::prealloc(size_t size)
 {
     REALM_ASSERT_RELEASE(is_attached());
 
 #if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
 
-    prealloc_if_supported(offset, size);
+    prealloc_if_supported(0, size);
 
 #else // Non-atomic fallback
 
-    if (int_add_with_overflow_detect(offset, size))
-        throw std::runtime_error("File size overflow");
-    if (get_size() < offset)
-        resize(offset);
+#if REALM_IOS
```

Comment:
Do you mean `REALM_PLATFORM_APPLE` here?

Reply:
Yes - I thought MacOS had posix_fallocate(), but apparently it doesn't. Since jenkins is acting weird it's a bit difficult to make the PR work everywhere.